The String() value of Nothing() should be in the grammar

rudle opened the pull request:
**What this PR does / why we need it**:

Changes the string representation of the Nothing() selector so that it may parse. 
Serializing Labels as strings is a supported operation, so it is imperative that
deserialization work also.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #38718

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access) 
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`. 
-->
```release-note
labels.Nothing().String() representation changed to something syntactically valid.
```

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/38719)
<!-- Reviewable:end -->

rudle committed:
The String() value of Nothing() should be in the grammar

Serializing Labels as strings is a supported operation, so it is imperative that
deserialization work also.

rudle commented:
I signed it! The CLA, that is.

Commit:
FormatLabels is consistent with the sentinel value for Nothing()

dims commented:
@k8s-bot ok to test

smarterclayton commented:
I'm not sure we want nothing to be allowed as an input value to the API - we deliberately do not support exclusion label selectors in the API today, and I'm hesitant to grow that.  On the other hand, I'd prefer anyone using label selectors outside of an API context at least have some way to distinguish nothing.

lavalamp commented:
I think this change is not right, doesn't this say `the label named "0" is not "0"`? Which does not actually select nothing. I would prefer to change the parsing section to parse `<null>`.
